.github/workflows/deploy-prod.yaml

```diff
@@ -129,28 +129,61 @@ jobs:
       env:
         STAGING_RELEASE: ${{ needs.get-staging-release.outputs.staging_release }}
 
-  copy-staging-release-to-prod:
-    runs-on: ubuntu-20.04
+  build-stage-packages-matrix:
+    runs-on: ubuntu-22.04
     needs:
      - get-staging-release
      - verify-staging-release-testgrid-result
+    outputs:
+      package: ${{ steps.set-matrix.output.package }}
     steps:
     - uses: actions/checkout@v3
       with:
         fetch-depth: 0
+    - name: Build Package Matrix
+      id: set-matrix
+      run: |
+        # Get list of packages associated with a staging release from s3:
+        # Exclude common and kurl-bin-utils packages => grep -vE "/common.*" | grep -vE "/kurl-bin-utils-*"
+        # Get raw strings NOT JSON encoded strings => jq -rc '.[]'
+        # Skip first line in output since its just the prefix and not a package name => tail -n +2
```

Comment:
it looks to me like this skips `staging/v2023.01.13-1-52ab8ebb/addons-gen.json`. why not grep -v it?

Reply:
we should exclude copying all *.json files?

What does `aws s3api list-objects-v2 --bucket "${S3_BUCKET}" --prefix "${STAGING_PREFIX}/${STAGING_RELEASE}" --query 'Contents[].Key' | jq -rc '.[]'` output for you in prod?
When I was testing in dev, I got the following output but note the first line is just the prefix w/o a key name so I decided to skip it:
```
rafael-staging/v2023.01.24-0-test/
rafael-staging/v2023.01.24-0-test/addons-gen.json
rafael-staging/v2023.01.24-0-test/aws-0.0.1.tar.gz
rafael-staging/v2023.01.24-0-test/calico-3.9.1.tar.gz
rafael-staging/v2023.01.24-0-test/common.tar.gz
rafael-staging/v2023.01.24-0-test/contour-0.14.0.tar.gz
rafael-staging/v2023.01.24-0-test/contour-1.0.1.tar.gz
rafael-staging/v2023.01.24-0-test/docker-18.09.8.tar.gz
rafael-staging/v2023.01.24-0-test/docker-19.03.4.tar.gz
rafael-staging/v2023.01.24-0-test/ekco-0.1.0.tar.gz
...
```

Reply:
I wasn't suggesting excluding json files. When I run in prod that is the first key. Perhaps if your intention is to exclude directories you should exclude keys with a trailing slash.

Reply:
I see. Don't blindly omit the first line just filter it out. I'll push a change for this.